src/input/mpegts/linuxdvb/linuxdvb_ddci.c

```diff
@@ -835,56 +889,84 @@ linuxdvb_ddci_open ( linuxdvb_ddci_t *lddci )
 }
 
 void
-linuxdvb_ddci_put ( linuxdvb_ddci_t *lddci, const uint8_t *tsb, int len )
-{
-  linuxdvb_ddci_wr_thread_buffer_put(&lddci->lddci_wr_thread, tsb, len );
+linuxdvb_ddci_put
+  ( linuxdvb_ddci_t *lddci, service_t *t, elementary_stream_t *st,
+    const uint8_t *tsb, int len )
+{
+  int_fast16_t pid = st ? st->es_pid : -1;
+
+  if (linuxdvb_ddci_cam_pid_filter(pid))
```

Comment:
As I wrote in previous PR, I don't think that it's a good idea to reorder MPEG-TS packets here.

Reply:
In the last days I talked to Klaus S. (VDR developer) and he does something similar. Only the CAT (required by the CAM) and Audio/Video packets and a faked (empty) EIT is send to the CAM in VDR. All the other PIDs of the stream go another route. This implies that they are not sent at the same position in the stream (sent from the streamdev plugin) as they were originally read from the tuner.

What problems do you see with a slightly sent earlier PAT, EIT, ...?
Is there any technical reason why you think it is a not good?

As you read above (comment from Daniel), clients like Kodi or VLC are working pretty good with the reordered stream. So it seems to me there is no requirement to keep the whole stream as is and in perfect order. Moreover, you don't know what the CAM does. In my opinion it will simply copy all not scrambled packets immediately to the output and descramble all the others. This needs time and the order on the CAM output is not the same as on the input, I guess.

In theory I could send all the packets through the CAM, even from different tuners (MTD), which requires to transform all the PIDs to service unique PIDs and map them back on the CAMs output. But I am sure it is not worth the effort, as the CAM will very likely reorder the packets anyway. And this approach will require a bit more bandwidth on the CI interface of the CAM, too.

Reply:
I don't think that the DDCI (or any descrambler) does any PID filtering and packet reordering. It just looks for the scrambled MPEG-TS packets and puts them descrambled to the output. So the information is not lost. It's also true that we reordered MPEG-TS packets in earlier tvh code, too (exactly like this - SI tables were passed to the output skipping the descrambling so that data for those mpegts PIDs were reordered in the output).

Yes, I agree that I the reordering works, but I'd prefer to keep the stream ASIS (contents) now. The bandwidth for those data is low, so the performance impact is also low. If you like, you may sent faked EIT (with markers to stored buffers to reconstruct the original stream on the output).

I just don't like the added code in linuxdvb_ddci_wr_thread_buffer_put() and this linuxdvb_ddci_put() fcn.